lib/activerecord/delay_touching.rb

```diff
@@ -86,18 +86,23 @@ def self.touch_records(attr, klass, records)
           column = column.to_s
           changes[column] = current_time
           records.each do |record|
-            next if record.destroyed?
+            next unless record.persisted? # A new record may have been rolled back
             record.instance_eval do
               write_attribute column, current_time
               @changed_attributes.except!(*changes.keys)
             end
           end
         end
 
-        klass.unscoped.where(klass.primary_key => records).update_all(changes)
+        updatable_records = records.select{|r| r.persisted?} # No need to touch unpersisted records
+
+        if updatable_records.present?
+          klass.unscoped.where(klass.primary_key => updatable_records).update_all(changes)
+        end
       end
       state.updated attr, records
       records.each do |record|
+        next unless record.persisted? # A new record may have been rolled back
```

Comment:
Does this now prevent a destroyed object from running `touch` to `commit` callbacks?